models/classes/UniqueId/Service/UniqueIdSetter.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2024 (original work) Open Assessment Technologies SA.
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\models\UniqueId\Service;
+
+use core_kernel_classes_Resource;
+use oat\generis\model\data\Ontology;
+use oat\tao\model\featureFlag\FeatureFlagCheckerInterface;
+use oat\tao\model\TaoOntology;
+use oat\tao\model\Translation\Service\ResourceMetadataRetriever;
+use oat\tao\model\UniqueId\Service\ResourceUniqueIdRetriever;
+use Psr\Log\LoggerInterface;
+use taoQtiTest_models_classes_QtiTestService;
+use Throwable;
+
+class UniqueIdSetter
```

Comment:
- [ ] This one seems exactly the same as for Items. Why did not we create it in tao-core